pkg/innerring/processors/container/common.go

```diff
@@ -66,16 +67,12 @@ func (cp *Processor) verifySignature(v signatureVerificationData) error {
 			return fmt.Errorf("decode session token: %w", err)
 		}
 
-		if !tok.VerifySignature() {
+		pub, ok := icrypto.VerifyTokenSignature(&tok)
+		if !ok {
 			return errors.New("invalid session token signature")
 		}
 
-		var signerPub neofsecdsa.PublicKeyRFC6979
-		if err = signerPub.Decode(tok.IssuerPublicKeyBytes()); err != nil {
-			return fmt.Errorf("invalid issuer public key: %w", err)
-		}
-
-		if user.NewFromECDSAPublicKey(ecdsa.PublicKey(signerPub)) != v.ownerContainer {
+		if user.NewFromECDSAPublicKey(*pub) != v.ownerContainer {
```

Comment:
It seems to be incompatible with verification scripts, although works at the current stage.

Reply:
I'll take another look, maybe I'll be able to move this check inside the function switching over schemes and unify it for all places